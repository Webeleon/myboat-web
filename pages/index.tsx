import MandatoryArmamentPage from './mandatoryArmament';

export const HomePage = (): JSX.Element => <MandatoryArmamentPage />;

export default HomePage;
