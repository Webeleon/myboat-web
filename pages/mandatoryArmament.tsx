import { navigationZone } from '../data/navigationZone'
import { useEffect, useState } from 'react'
import { mandatoryArmament } from '../data/mandatoryArmament'
import styles from '../styles/pages/armament.module.scss'
import {
  getMandatoryArmamentFromLocalStorage,
  mandatoryArmamentInLocalStorage,
  setMandatoryArmamenetItemStatus,
} from '../localStorage/mandatoryArmament'

export const MandatoryArmamentPage = (): JSX.Element => {
  const [zone, setZone] = useState<navigationZone>(navigationZone.COASTAL)
  const [equipement, setEquipement] = useState<mandatoryArmamentInLocalStorage>(
    {}
  )

  useEffect(() => {
    setEquipement(getMandatoryArmamentFromLocalStorage())
  }, [])

  return (
    <div className="app">
      <div className={styles.zoneSelectorContainer}>
        <select
          name="navigation-zone"
          value={zone}
          onChange={(event) => setZone(event.target.value as navigationZone)}
        >
          {Object.values(navigationZone).map((z) => (
            <option key={z} value={z}>
              {z}
            </option>
          ))}
        </select>
      </div>
      <ul className="stripped">
        {mandatoryArmament
          .filter((equipementItem) => equipementItem.zones.includes(zone))
          .map((equipementItem) => (
            <li key={equipementItem.name}>
              <span>{equipementItem.name}</span>
              <input
                type="checkbox"
                checked={equipement[equipementItem.name]}
                onChange={(e) => {
                  setEquipement({
                    ...equipement,
                    [equipementItem.name]: e.target.checked,
                  })
                  setMandatoryArmamenetItemStatus(
                    equipementItem.name,
                    e.target.checked
                  )
                }}
              />
            </li>
          ))}
      </ul>
    </div>
  )
}

export default MandatoryArmamentPage
