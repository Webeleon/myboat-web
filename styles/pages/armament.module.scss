.zoneSelectorContainer {
  padding: 1em 0;
  display: flex;
  justify-content: center;

  select {
    padding: 1em;
  }
}
