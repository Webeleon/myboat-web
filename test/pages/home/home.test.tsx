import React from 'react';
import { render } from '../../testUtils';
import { HomePage } from '../../../pages';

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<HomePage />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
