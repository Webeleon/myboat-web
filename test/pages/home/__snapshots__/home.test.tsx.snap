// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="zoneSelectorContainer"
    >
      <select
        name="navigation-zone"
      >
        <option
          value="basic"
        >
          basic
        </option>
        <option
          value="coastal"
        >
          coastal
        </option>
        <option
          value="semi_offshore"
        >
          semi_offshore
        </option>
        <option
          value="offshore"
        >
          offshore
        </option>
      </select>
    </div>
    <ul
      class="stripped"
    >
      <li>
        <span>
          Équipement individuel de flottabilité
        </span>
        <input
          type="checkbox"
        />
      </li>
      <li>
        <span>
          Dispositif lumineux
        </span>
        <input
          type="checkbox"
        />
      </li>
      <li>
        <span>
          Moyens mobiles de lutte contre l'incendie
        </span>
        <input
          type="checkbox"
        />
      </li>
      <li>
        <span>
          Dispositif d'assèchement manuel
        </span>
        <input
          type="checkbox"
        />
      </li>
      <li>
        <span>
          Dispositif de remorquage
        </span>
        <input
          type="checkbox"
        />
      </li>
      <li>
        <span>
          Ligne de mouillage (si masse lège ≥ 250 kgs)
        </span>
        <input
          type="checkbox"
        />
      </li>
      <li>
        <span>
          Annuaire des marées
        </span>
        <input
          type="checkbox"
        />
      </li>
      <li>
        <span>
          Pavillon national (hors eaux territoriales)
        </span>
        <input
          type="checkbox"
        />
      </li>
      <li>
        <span>
          Dispositif de repérage et d'assistance pour personne à la mer
        </span>
        <input
          type="checkbox"
        />
      </li>
      <li>
        <span>
          3 feux rouges à main
        </span>
        <input
          type="checkbox"
        />
      </li>
      <li>
        <span>
          Compas magnétique
        </span>
        <input
          type="checkbox"
        />
      </li>
      <li>
        <span>
          Cartes marines officielles
        </span>
        <input
          type="checkbox"
        />
      </li>
      <li>
        <span>
          Règlement international pour prévenir les abordages en mer
        </span>
        <input
          type="checkbox"
        />
      </li>
      <li>
        <span>
          Description du système de balisage
        </span>
        <input
          type="checkbox"
        />
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
